Source_Code/TM_WebSites/Website_3.5/TBot/Admin/Publish_Data.cshtml

```diff
@@ -0,0 +1,66 @@
+﻿
+<h4>Publish TEAM Mentor Data</h4>
+
+<script type="text/javascript">
+    var tbot = angular.module("tbot");
+
+    tbot.controller("TMPublishData", function($scope,$http, $timeout)
+        {
+            $scope.phase = "show publish button";
+
+            $scope.publishData = function()
+                {
+                    $scope.message_userDataReload = "Publishing TEAM Mentor Data";
+                    $scope.startPublishing(); // so that the user can read the first message
+                };
+         
+            $scope.startPublishing = function()
+                {
+                    $scope.phase = 'publishing';
+                    $http.get("/rest/admin/publish_Data")
+                    .success(function(data) {
+                            if (data =="true") {
+                                 return $scope.publishComplete();
+                            }else return $scope.publishRestarting();
+                        })
+                      .error($scope.publishRestarting);
+                };
+
+            $scope.publishComplete = function()
+                {
+                    $scope.phase = 'publish complete';
+                    $scope.message_userDataReload = "Data Published!.";
+                };
+            $scope.publishRestarting = function()
+                {
+                    $scope.phase = 'Publish failed';
+                };
+        });
```

Comment:
Also can you refactor the .success into a function like you did with publishComplete and publishRestarting (it looks much better)